lib/samson/secrets/shared/list_recursive.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Add recursive listing which was rejected by vault-ruby see https://github.com/hashicorp/vault-ruby/pull/118
+# using a monkey patch so `vault_action :list_recursive` works in the backend
+module Samson
+  module Secrets
+    module Shared
+      module ListRecursive
+        # Limit scope to logical after subclass call
```

Comment:
idk why this is needed, could you explain more or try to remove ?

Reply:
It's needed because the wrappers are inheriting from the classes they are wrapping, so for example `VaultLogicalWrapper` inherits from `Vault::Logical`. `VaultLogicalWrapper` has `list` and `list_recursive` defined to provide the unified interface for calling vault functions in Samson code. When we call `VaultLogicalWrapper#list_recursive`, that will call `super()`, at which point that method will call `list_recursive` and `list` multiple times as it reads secrets. We want these calls to be `Logical`'s methods, not the wrappers, so I use the aliasing to limit scope to the parent once we enter that method.

Reply:
add something like that as inline comment plz :)

Reply:
Will do 👍